RecoBTag/DeepFlavour/plugins/DeepFlavourJetTagProducer.cc

```diff
@@ -0,0 +1,261 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/DeepFormats/interface/DeepFlavourTagInfo.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "RecoBTag/DeepFlavour/interface/tensor_fillers.h"
+
+// Declaration of the data structure that is hold by the edm::GlobalCache.
+// In TensorFlow, the computational graph is stored in a stateless meta graph object which can be
+// shared by multiple session instances which handle the initialization of variables related to the
+// meta graph. Following this approach in CMSSW, a meta graph should be stored in a GlobalCache
+// which can be accesses by sessions owned by multiple stream module copies. Instead of using only
+// the plain meta graph, we make use of a Cache struct that can be extended in the future if nedded.
+// In addition, the meta graph is protected via std::atomic, which should not affect the performance
+// as it is only accessed in the module constructor and not in the actual produce loop.
+struct Cache {
+  Cache() : metaGraph(nullptr)
+  {
+  }
+
+  std::atomic<tf::MetaGraphDef*> metaGraph;
+};
+
+class DeepFlavourJetTagProducer : public edm::stream::EDProducer<edm::GlobalCache<Cache>> {
```

Comment:
I'm now realizing that we also have DeepFlavourJetTagsProducer, which may be confusing to keep track of.
Please rename this to e.g DeepFlavourTFJetTagsProducer  (with TF for tensorflow) or something in line.

Reply:
I will change, it is definitely misleading

Reply:
changed in https://github.com/cms-sw/cmssw/pull/19893/commits/73d6cacd8b29db246d053d795014163777b6620e and https://github.com/cms-sw/cmssw/pull/19893/commits/b813f0c10ffffc1732a5d81e2aa99c74cf54c720